src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1516,6 +1525,291 @@ def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=Non
+                        time.sleep(10)
+                        break
+
+                    network_config_array += json.loads('[' + network_config + ']')
+                    network_status_object = json.loads(network_status)
+                    network_status_array += format_diag_status(network_status_object)
+                    ready_nodes[node_name] = True
+                except subprocess.CalledProcessError as err:
+                    raise CLIError(err.output)
+
+    print()
+    if network_config_array:
+        print("Below are the network configuration for each node: ")
+        print()
+        print(tabulate(network_config_array, headers="keys", tablefmt='simple'))
+        print()
+    else:
+        logger.warning("Could not get network config. "
+                       "Please run 'az aks kanalyze' command later to get the analysis results.")
+
+    if network_status_array:
+        print("Below are the network connectivity results for each node:")
+        print()
+        print(tabulate(network_status_array, headers="keys", tablefmt='simple'))
+    else:
+        logger.warning("Could not get networking status. "
+                       "Please run 'az aks kanalyze' command later to get the analysis results.")
+
+def format_diag_status(diag_status):
+    for diag in diag_status:
+        if diag["Status"]:
+            if "Error:" in diag["Status"]:
+                diag["Status"] = f'{colorama.Fore.RED}{diag["Status"]}{colorama.Style.RESET_ALL}'
+            else:
+                diag["Status"] = f'{colorama.Fore.GREEN}{diag["Status"]}{colorama.Style.RESET_ALL}'
+
+    return diag_status
+
+
+def collect_logs(cmd, resource_group_name, name, storage_account_name=None,
+                sas_token=None,
+                readonly_sas_token=None,
+                kube_context=None,
+                kube_config=None,
+                ):
+    colorama.init()
+
+    if not which('kubectl'):
+        raise CLIError('Can not find kubectl executable in PATH')
+
+    readonly_sas_token = readonly_sas_token.strip('?')
+
+    from knack.prompting import prompt_y_n
+
+    print()
+    print('This will deploy a daemon set to your cluster to collect logs and diagnostic information and '
+          f'save them to the storage account '
+          f'{colorama.Style.BRIGHT}{colorama.Fore.GREEN}{storage_account_name}{colorama.Style.RESET_ALL} as '
+          f'outlined in {format_hyperlink("http://aka.ms/AKSPeriscope")}.')
+    print()
+    print('If you share access to that storage account to Azure support, you consent to the terms outlined'
+          f' in {format_hyperlink("http://aka.ms/DiagConsent")}.')
+    print()
+    if not prompt_y_n('Do you confirm?', default="n"):
+        return
+    print()
+    print("Getting credentials for cluster %s " % name)
+    path=os.path.join(os.path.expanduser('~'), '.kube', 'config')
+    # Setting kubeconfig
+    # Send cloud information to telemetry
+    send_cloud_telemetry(cmd)
```

Comment:
Not needed.

Reply:
I had forgotten, to add this just at the beginning of the troubleshoot command function. Please, move it there. Thanks.

Reply:
Sure, I have moved the send_cloud_telemetry(cmd) to the beginning of the troubleshoot command.